src/cljc/rems/common/form.cljc

```diff
@@ -62,3 +79,297 @@
                                                  :visibility/values ["yes" "definitely"]}}
                              {"1" "definitely"}))))
 
+(defn- validate-text-field [m key]
+  (when (str/blank? (get m key))
+    {key :t.form.validation/required}))
+
+(def field-types #{:attachment :date :description :email :header :label :multiselect :option :text :texta})
+
+(defn- validate-field-type [m]
+  (let [type (keyword (get m :field/type))]
+    (cond
+      (not type)
+      {:field/type :t.form.validation/required}
+
+      (not (contains? field-types type))
+      {:field/type :t.form.validation/invalid-value})))
+
+(defn- validate-localized-text-field [m key languages]
+  {key (apply merge (mapv #(validate-text-field (get m key) %) languages))})
+
+(defn- validate-optional-localized-field [m key languages]
+  (let [validated (mapv #(validate-text-field (get m key) %) languages)]
+    ;; partial translations are not allowed
+    (when (not-empty (remove identity validated))
+      {key (apply merge validated)})))
+
+(def ^:private max-length-range [0 32767])
+
+(defn- validate-max-length [max-length]
+  {:field/max-length
+   (cond (nil? max-length)
+         nil
+
+         (int? max-length)
+         (when-not (<= (first max-length-range) max-length (second max-length-range))
```

Comment:
it might make this function a bit shorter if you defined:
```
(let [parsed (if (int? max-length) max-length (parse-int max-length)]
  ...)

Reply:
Yes!